doc/nrf/gs_testing.rst

```diff
@@ -120,3 +120,74 @@ To connect to the nRF9160-based kit with LTE Link Monitor, perform the following
+#. Once both sessions are established, execute the code on the application core.
+
+   The startup code releases the ``NETWORK.FORCEOFF`` signal to start the network core and allocates the necessary GPIO pins for it.
+#. Start code execution on the network core in the other debug session.
+
+If you want to reset the network core while debugging, make sure to first reset the application core and execute the code.
+
+.. note::
+  Debugging firmware on the application core in the non-secure domain (that is, when using the ``_ns`` build target), is currently not supported.
+
+Debug configuration
+===================
+
+The following Kconfig options are used by the |VSC| debugger in the |nRFVSC|:
+
+* :kconfig:option:`CONFIG_DEBUG_OPTIMIZATIONS` set to ``y``
+* :kconfig:option:`CONFIG_DEBUG_THREAD_INFO` set to ``y``
+
+Debug logging
+-------------
+
+Some protocols, applications, and samples may require additional Kconfig options to be set when debugging.
+Among these are logging backend configuration options that need to be set for the logging system to provide debug logs.
+For example, nRF9160's :ref:`nRF Cloud MQTT multi-service <nrf_cloud_mqtt_multi_service>` sample will start printing debug logs after you enable :ref:`CONFIG_MQTT_MULTI_SERVICE_LOG_LEVEL_DBG <CONFIG_MQTT_MULTI_SERVICE_LOG_LEVEL_DBG>`.
+
+For more information, read the :ref:`ug_logging` user guide.
+To list all debug-related Kconfig options in the |NCS|, search for "DEBUG" on the :ref:`kconfig-search` page.
+
+Debug build types
+-----------------
+
+Some applications and samples provide a specific build type that enables additional debug functionalities.
+You can select these build types when you are :ref:`configuring the build settings <gs_modifying_build_types>`.
+
+Debug libraries
+---------------
+
+The |NCS| also provides several libraries and drivers for debugging different components.
+
+* When working with the :ref:`SEGGER J-Link with the RTT feature <testing_rtt_connect>`, you can use the :ref:`lib_eth_rtt`, useful to handle data transfer.
+
+* The Thread protocol offers :ref:`pre-built libraries with debug symbols <thread_ug_feature_updating_libs>`.
+
+* The Zigbee protocol offers :ref:`lib_zigbee_osif` and :ref:`lib_zigbee_shell` with custom Kconfig options that you can set for debugging.
+
+Debug tools
+===========
+
+The main recommended tool for debugging in the |NCS| is the `GNU Project Debugger`_ (GDB tool).
+
+* When working with the |nRFVSC|, use the GDB tool after adding the required Kconfig options to the :file:`prj.conf` file.
+* If you are working from the command line, you can use west with the GDB tool.
+  For details, read the :ref:`Debugging with west debug <zephyr:west-debugging>` section on the :ref:`zephyr:west-build-flash-debug` page in the Zephyr documentation.
+
+A useful tool for debugging the communication over Bluetooth and mesh networking protocols, such as :ref:`ug_thread` or :ref:`ug_zigbee`, is the `nRF Sniffer for 802.15.4`_.
```

Comment:
`The nRF Sniffer allows you to look into data exchanged over the air between devices.`